punishments.js

```diff
@@ -389,7 +416,7 @@ setImmediate(() => {
  * @param {User} user
  * @param {Punishment} punishment
  * @param {?Set<string>} recursionKeys
- * @return {?Array}
+ * @return {Array | undefined}
```

Comment:
`Array` is not really a good look here. At least use `any[]` to make it clearer you're opting out of typechecking. Honestly, `@typedef {any[]} PunishmentRow`; will make it easier to actually properly type punishmentrows later.

Reply:
Okay, I only realized now that these aren't actually `PunishmentRow`s but just regular arrays of users.